### PR TITLE
Add retract case for Go 1.16 go.mod files

### PR DESCRIPTION
**Pull request by jakeryang**
Addresses #69 

This case should behave exactly like the `exclude` case.

Should probably revisit once [Go 1.16 release notes](https://tip.golang.org/doc/go1.16#go-command) include more information.

### CODE REVIEW

File: extractor/src/extractors/GoModExtractor.ts [hunk 1]
```diff
@@ -84,7 +84,8 @@ export default class GoModExtractor implements Extractor {
                         }
                     }
                     break;
-
+                
+                case "retract":
```
Comment (mjpitz) [inline extractor/src/extractors/GoModExtractor.ts:88]: We should add an associated test case for this as well to ensure the line is parsed properly.

**Review by mjpitz — CHANGES_REQUESTED** [review]

File: extractor/src/extractors/GoModExtractor.ts (hunk 1 above)
Comment (jakeryang) [reply extractor/src/extractors/GoModExtractor.ts:88]: I agree. However I'm having trouble running the extractor test cases on my local machine by doing `npm test` in the extractor directory. The test cases run but a couple fail. I also tried building and running the Dockerfile.dev in the same directory and I got the same results. Is there something that I'm not doing correctly? 

Once I figure this out I'll look more into the test cases.

Comment (jakeryang) [reply extractor/src/extractors/GoModExtractor.ts:88]: I ran the test cases with `npm test -- -u` instead and they seem to work now. I'll get to work on this.

Comment (jakeryang) [reply extractor/src/extractors/GoModExtractor.ts:88]: Realized that I was updating the Snapshots so of course it worked... 

Now I'm getting an empty array of dependencies for GoModExtractor, even with the `retract` case removed.

**Review by mjpitz — CHANGES_REQUESTED** [review]
See my comment [here](https://github.com/depscloud/depscloud/issues/69#issuecomment-705080220) about updating the default block instead of treating it as an exclusion. This was a change after learning more about its use. This will allow us to more gracefully handle the case while we figure out how to properly model this.

File: extractor/src/extractors/GoModExtractor.ts [hunk 2]
```diff
@@ -84,16 +84,20 @@ export default class GoModExtractor implements Extractor {
...
                     }
                     break;
-
+                
+                case "//":
+                    break;
+                
                 default:
-                    throw new Error(`parse error: unsupported directive: ${directive}`);
+                    console.debug(`parse error: unsupported directive: ${directive}`);
```
Comment (mjpitz) [inline extractor/src/extractors/GoModExtractor.ts:100]: Don't use console. This project uses log4js:

https://github.com/depscloud/depscloud/blob/main/extractor/src/main.ts#L4
https://github.com/depscloud/depscloud/blob/main/extractor/src/main.ts#L22

**Comment by mjpitz** [comment]
One final set of changes (move from console to the project logging library)

**Review by mjpitz — APPROVED** [review]

**Pull request merged**